src/strategy/periodic.rs

```diff
@@ -41,15 +58,57 @@ impl StrategyPeriodic {
             n => log::trace!("periodic updates, weekly calendar length: {} minutes", n),
         };
 
-        let strategy = Self { schedule: calendar };
+        let strategy = Self {
+            schedule: calendar,
+            time_zone: tz,
+        };
         Ok(strategy)
     }
 
+    /// Get the time zone from `periodic` strategy config.
+    fn get_time_zone(cfg: &inputs::PeriodicInput) -> Fallible<Tz> {
+        let tz;
+        if &cfg.time_zone == "localtime" {
+            // Use `read_link` instead of `exists()` because we only want to check for
+            // the existence of the `/etc/localtime` symlink, not whether it points to
+            // a valid location. (`read_link` returns an error if symlink doesn't exist)
```

Comment:
To add some context as to why I used `read_link()` here, we only want to default to `UTC` if `/etc/localtime` doesn't exist, and not if the location is invalid, in which case `Tz::local()` will return an Error. This is to prevent surprising users with a UTC time zone if they _think_ they're using `localtime`, but accidentally symlinked an invalid time zone when trying to configure their system's local time.